core/src/main/java/org/hisp/dhis/android/core/arch/api/filters/internal/SingleValueFilter.kt

```diff
@@ -25,48 +25,39 @@
  *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
  *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
+package org.hisp.dhis.android.core.arch.api.filters.internal
 
-package org.hisp.dhis.android.core.arch.api.filters.internal;
+import org.hisp.dhis.android.core.arch.api.fields.internal.Field
 
-import com.google.auto.value.AutoValue;
-
-import org.hisp.dhis.android.core.arch.api.fields.internal.Field;
-
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Iterator;
+internal class SingleValueFilter<T, K> (
+    override val field: Field<T, K>,
+    override val operator: String,
+    override val values: Collection<String>,
+) : Filter<T, K> {
+    override fun generateString(): String {
+        return "${field.name()}:$operator:${values.first()}"
```

Comment:
This method will always have a collection with a single value. But the constructor is internal so it is possible that other parts of the code create instances of this class with an empty list. Could it be possible to make the constructor internal?